aws/resource_aws_globalaccelerator_endpoint_group.go

```diff
@@ -0,0 +1,360 @@
+	conn := meta.(*AWSClient).globalacceleratorconn
+
+	opts := &globalaccelerator.UpdateEndpointGroupInput{
+		EndpointGroupArn: aws.String(d.Id()),
+	}
+
+	if v, ok := d.GetOk("health_check_interval_seconds"); ok {
+		opts.HealthCheckIntervalSeconds = aws.Int64(int64(v.(int)))
+	}
+
+	if v, ok := d.GetOk("health_check_path"); ok {
+		opts.HealthCheckPath = aws.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("health_check_port"); ok {
+		opts.HealthCheckPort = aws.Int64(int64(v.(int)))
+	}
+
+	if v, ok := d.GetOk("health_check_protocol"); ok {
+		opts.HealthCheckProtocol = aws.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("threshold_count"); ok {
+		opts.ThresholdCount = aws.Int64(int64(v.(int)))
+	}
+
+	if v, ok := d.GetOk("traffic_dial_percentage"); ok {
+		opts.TrafficDialPercentage = aws.Float64(v.(float64))
+	}
+
+	if v, ok := d.GetOk("endpoint_configurations"); ok {
+		opts.EndpointConfigurations = resourceAwsGlobalAcceleratorEndpointGroupExpandEndpointConfigurations(v.([]interface{}))
+	} else {
+		opts.EndpointConfigurations = []*globalaccelerator.EndpointConfiguration{}
+	}
+
+	log.Printf("[DEBUG] Update Global Accelerator endpoint group: %s", opts)
+
+	_, err := conn.UpdateEndpointGroup(opts)
+	if err != nil {
+		return fmt.Errorf("Error updating Global Accelerator endpoint group: %s", err)
+	}
+
+	acceleratorArn, err := resourceAwsGlobalAcceleratorListenerParseAcceleratorArn(d.Id())
+	if err != nil {
+		return err
+	}
+
+	// Creating an endpoint group triggers the accelerator to change status to InPending
+	stateConf := &resource.StateChangeConf{
+		Pending: []string{globalaccelerator.AcceleratorStatusInProgress},
+		Target:  []string{globalaccelerator.AcceleratorStatusDeployed},
+		Refresh: resourceAwsGlobalAcceleratorAcceleratorStateRefreshFunc(conn, acceleratorArn),
+		Timeout: d.Timeout(schema.TimeoutUpdate),
+	}
+
+	log.Printf("[DEBUG] Waiting for Global Accelerator endpoint group (%s) availability", d.Id())
+	_, err = stateConf.WaitForState()
+	if err != nil {
+		return fmt.Errorf("Error waiting for Global Accelerator endpoint group (%s) availability: %s", d.Id(), err)
+	}
+
+	return resourceAwsGlobalAcceleratorEndpointGroupRead(d, meta)
+}
+
+func resourceAwsGlobalAcceleratorEndpointGroupDelete(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).globalacceleratorconn
+
+	opts := &globalaccelerator.DeleteEndpointGroupInput{
+		EndpointGroupArn: aws.String(d.Id()),
+	}
+
+	_, err := conn.DeleteEndpointGroup(opts)
+	if err != nil {
+		if isAWSErr(err, globalaccelerator.ErrCodeEndpointGroupNotFoundException, "") {
+			return nil
+		}
+		return fmt.Errorf("Error deleting Global Accelerator endpoint group: %s", err)
+	}
+
+	acceleratorArn, err := resourceAwsGlobalAcceleratorListenerParseAcceleratorArn(d.Id())
+	if err != nil {
+		return err
+	}
+
+	// Deleting an endpoint group triggers the accelerator to change status to InPending
```

Comment:
This logic should reuse the available function (albeit slightly poor naming): `resourceAwsGlobalAcceleratorAcceleratorWaitForState`

```go
err = resourceAwsGlobalAcceleratorAcceleratorWaitForState(conn, acceleratorArn)
if err != nil {
	return err
}
```